### PR TITLE
Ria 1152 & 1153 generate decision and reasons

### PR DESCRIPTION
**Pull request by t80**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/functionalTest/resources/scenarios/RIA-1152-&-1153-generate-hearing-ready-bundle.json [hunk 1]
```diff
@@ -0,0 +1,55 @@
+{
```
Comment (raghera) [inline src/functionalTest/resources/scenarios/RIA-1152-&-1153-generate-hearing-ready-bundle.json]: Think this filename should not include `generate-hearing-ready-bundle`

File: src/functionalTest/resources/scenarios/RIA-706-upload-additional-evidence-submit.json [hunk 2]
```diff
@@ -60,7 +60,7 @@
     "caseData": {
       "template": "minimal-appeal-submitted.json",
       "replacements": {
-        "additionalDocuments": [
+        "additionalEvidenceDocuments": [
```
Comment (raghera) [inline src/functionalTest/resources/scenarios/RIA-706-upload-additional-evidence-submit.json]: did we change an existing field name here?  Why was that?  Just wondering if it affects any existing functionality.

File: src/integrationTest/java/uk/gov/hmcts/reform/iacaseapi/Service2ServiceEnabledIntegrationTest.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+import static com.github.tomakehurst.wiremock.client.WireMock.*;
+
+import com.github.tomakehurst.wiremock.junit.WireMockRule;
+import org.junit.Before;
+import org.junit.Rule;
+import org.springframework.test.context.TestPropertySource;
+
+@TestPropertySource(properties = {"S2S_URL=http://127.0.0.1:8990"})
+public abstract class Service2ServiceEnabledIntegrationTest extends SpringBootIntegrationTest {
```
Comment (raghera) [inline src/integrationTest/java/uk/gov/hmcts/reform/iacaseapi/Service2ServiceEnabledIntegrationTest.java]: Did you mean to include this here?  Probably from the Asylum-case-refactor branch to be honest as this is rebased on top.

File: src/integrationTest/java/uk/gov/hmcts/reform/iacaseapi/SpringBootIntegrationTest.java [hunk 4]
```diff
@@ -0,0 +1,15 @@
+package uk.gov.hmcts.reform.iacaseapi;
+
+import org.junit.runner.RunWith;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.web.server.LocalServerPort;
+import org.springframework.test.context.junit4.SpringRunner;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest(classes = Application.class, webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
```
Comment (raghera) [inline src/integrationTest/java/uk/gov/hmcts/reform/iacaseapi/SpringBootIntegrationTest.java]: Not sure we need this for this feature.

File: src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/AppellantNameForDisplayFormatter.java [hunk 5]
```diff
@@ -23,34 +24,35 @@ public boolean canHandle(
     }
 
     public PreSubmitCallbackResponse<AsylumCase> handle(
-        PreSubmitCallbackStage callbackStage,
-        Callback<AsylumCase> callback
+            PreSubmitCallbackStage callbackStage,
```
Comment (raghera) [inline src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/AppellantNameForDisplayFormatter.java]: Formatting again seems to have gone awry again.

File: src/functionalTest/resources/scenarios/RIA-706-upload-additional-evidence-submit.json (hunk 2 above)
Comment (t80) [reply src/functionalTest/resources/scenarios/RIA-706-upload-additional-evidence-submit.json]: it seemed to have been changed in the spreadsheet code.  This test was failing.

File: src/integrationTest/java/uk/gov/hmcts/reform/iacaseapi/Service2ServiceEnabledIntegrationTest.java (hunk 3 above)
Comment (t80) [reply src/integrationTest/java/uk/gov/hmcts/reform/iacaseapi/Service2ServiceEnabledIntegrationTest.java]: So forgot to mention, original I wanted to introduce an integration test using the new framework, but there was something odd going on with an endpoint /lease that meant it was failing on jenkins but passing locally.  I have a suspicion that I might have left the CCD container running when testing locally, so it was using that.  I didnt have time work on them further so left the framework in place as its only in test.  Will look into this as part of the next story and evolve further.

File: src/functionalTest/resources/scenarios/RIA-1152-&-1153-generate-hearing-ready-bundle.json (hunk 1 above)
Comment (t80) [reply src/functionalTest/resources/scenarios/RIA-1152-&-1153-generate-hearing-ready-bundle.json]: sorted

File: src/integrationTest/java/uk/gov/hmcts/reform/iacaseapi/Service2ServiceEnabledIntegrationTest.java (hunk 3 above)
Comment (t80) [reply src/integrationTest/java/uk/gov/hmcts/reform/iacaseapi/Service2ServiceEnabledIntegrationTest.java]: as discussed

**Pull request merged**